packages/fxa-auth-server/lib/sentry.js

```diff
@@ -256,18 +257,56 @@ async function configureSentry(server, config, processName = 'key_server') {
  *                use the underlying jse_cause error if possible.
  */
 function ignoreErrors(error) {
-  if (!error) {
-    return;
+  // Short circuit for null / undefined errors.
+  if (error == null) {
+    return true;
+  }
+
+  // Check if the error has an underlying cause. This will drill down until we
+  // leave the vError chain, and return the underlying error.
+  const cause = getRootCause(error);
+
+  // If the cause was an AppError, check if it is ignorable.
+  if (cause instanceof AppError) {
+    return (
+      IGNORED_ERROR_NUMBERS.includes(cause.errno) ||
+      determineStatusCode(cause) < 500
+    );
+  }
+
+  // If the cause was null, but the top level error was an AppError, check if the top level error
+  // can be ignored.
+  if (cause == null && error instanceof AppError) {
+    return (
+      IGNORED_ERROR_NUMBERS.includes(error.errno) ||
+      determineStatusCode(error) < 500
+    );
   }
 
-  // Prefer jse_cause, but fallback to top level error if needed
-  const statusCode =
-    determineStatusCode(error.jse_cause) || determineStatusCode(error);
+  // Otherwise, don't ignore the error. Either the cause wasn't an AppError,
+  // or the cause was null and the top level error wasn't an AppError.
+  return false;
+}
 
-  const errno = error.jse_cause?.errno || error.errno;
+/**
+ * Drill down the vError chain to find the underlying error.
+ * @param {Error} error
+ * @returns {Error|null}
+ */
+function getRootCause(error) {
+  const cause =
+    error?.jse_cause != null
+      ? error.jse_cause
+      : typeof error?.cause === 'function'
+      ? error.cause()
+      : null;
```

Comment:
This chaining is a bit difficult to read for me.

Reply:
I'll expanded it to if statements